### DISCUSSION TITLE
Newest version Lychee does not allow create new folder.

### DISCUSSION THREAD

**Discussion by bussekas** [LycheeOrg/Lychee#2422] [Q&A]
Hi

After latest upgrade version to 5.3 when i try to create new folder i get error:
_Forbidden
You don't have permission to access this resource.

Apache / ZoneOS Server_

Is there some problem with permissions or what i can do to fix it?

**Comment by ildyria** [comment]
That's weird. Are you admin user ?

**Reply by bussekas** [reply]
Yes, i upgrade from 4.xx to 5.xx and i'm using db for storing data.

**Reply by ildyria** [reply]
I will need a bit more info than that.
More specifically the content of your diagnostic page.

**Reply by bussekas** [reply]
Can i set on debug mode some how?

<img width="803" alt="Screenshot 2024-05-02 213918" src="https://github.com/LycheeOrg/Lychee/assets/28824069/dec1c7b0-2bfc-419e-b8d0-3024beef0f75">

when i click + then error arrive. maybe problem is that i can add only 775 not 2755?

diagnostic data:
Warning: /data******************************************ublic/uploads/ has permissions 0755, but should have 2775
        Warning: /data******************************************ublic/uploads/original has permissions 0755, but should have 2775
        Warning: /data******************************************ublic/uploads/original/bf has permissions 0755, but should have 2775
        Warning: /data******************************************ublic/uploads/original/bf/b5 has permissions 0755, but should have 2775
        Warning: /data******************************************ublic/uploads/original/bf/e1 has permissions 0755, but should have 2775
        Warning: 25611 more directories with wrong permissions
        Warning: Dropbox import not working. dropbox_key is empty.
        Warning: You may experience problems when uploading a photo of large size or handling many/large albums. Take a look in the FAQ for details.
        Info: Latest version of PHP is 8.3
         Foreign key: access_permissions.user_id     → users.id             : CASCADE
         Foreign key: albums.parent_id               → albums.id            : RESTRICT
         Foreign key: albums.id                      → base_albums.id       : RESTRICT
         Foreign key: albums.cover_id                → photos.id            : CASCADE
         Foreign key: base_albums.owner_id           → users.id             : RESTRICT
         Foreign key: jobs_history.owner_id          → users.id             : RESTRICT
         Foreign key: oauth_credentials.user_id      → users.id             : CASCADE
         Foreign key: photos.album_id                → albums.id            : RESTRICT
         Foreign key: photos.owner_id                → users.id             : RESTRICT
         Foreign key: size_variants.photo_id         → photos.id            : RESTRICT
         Foreign key: sym_links.size_variant_id      → size_variants.id     : RESTRICT
         Foreign key: tag_albums.id                  → base_albums.id       : RESTRICT

**Reply by ildyria** [reply]
nah this is not related to the permissions set.

This is supposed to open a drop down menu and it seems that Livewire is not working well. The weirdest part is that you don't complain about not being able to open albums/details.

To me this is could also be a miss configuration on your APACHE side rather than on Lychee.

Could you try to clear the cache ?
`php artisan optimize:clear`

**Reply by bussekas** [reply]
I must say that it was my first upgrade what almost did success, before i have download .zip and update and there was always some error so i turn it old version back or create clean install from .zip and then upload all data again. 

this week was first time when i try to upload new gallery and then i notice this problem.

site is hosted by zone.ee hosting. what value is behind this + mark what i can't opened after upgrade?

for php artisan optimize:clear i should access over ssl to server probably?

**Reply by bussekas** [reply]
I notice that .env file from new download vs v4 is kind different. should it somehow update this file or just leave it as is?

in log i notice error latest version:
![Screenshot 2024-05-03 230821](https://github.com/LycheeOrg/Lychee/assets/28824069/3c07a41c-0ac3-41c3-b6c6-52c8b4ca0b50)

is it ok?

also i did latest lychee download and just add from old set upload files, .env and user.css and still get same error. also did create new db scheme from old set with new db user but it did not change also anything.

So is there an option that i can create new install and only update uploads files with db from old set?

**Reply by bussekas** [reply]
I made in sub domain clean install and it did work well. there is no error but how then i can move all my existing data into new install without any future problem?

**Reply by ildyria** [reply]
You just need to move the **upload folder** and your database (I assume you use mysql).
Something like:
```
/usr/bin/mysqldump -u <username> --password=<password> <dbname> > backup.sql
```
and then reimport
```
/usr/bin/mysql -u <username> --password=<password> <dbname> < backup.sql
```

You may need to apply some migrations:
```
php artisan:migrate
```
Note that this will override all your data.

**Reply by bussekas** [reply]
So i made some test. clean install in sub domain and there buttons did work and appear. now i take this clean install and put it into main domain (change .env file) and buttons just lost after that.

**Reply by ildyria** [reply]
Update. Given that you have some config error, I would not recommend doing the full backup but more of a targeted one.
I assume that you still have access to the broken install. I advise you to:
apply the migration on the DB (to be sure you are on the latest version)

And just dump the following tables:
```
users
access_permissions
albums
base_albums
oauth_credentials
photos
size_variants
tag_albums
webauthn_credentials
```
Those are the table which contains the data.

**Reply by ildyria** [reply]
> So i made some test. clean install in sub domain and there buttons did work and appear. now i take this clean install and put it into main domain (change .env file) and buttons just lost after that.

What the hell.

**Reply by ildyria** [reply]
Would the apache config be different between subdomain and full?

**Reply by bussekas** [reply]
what i don't have made because i don't have access to ssh is _php artisan:migrate._ i understand that this should be made over ssh.

so what i should try next is make db dump from tables only... and then take from .zip all files and but into domain. then should add/write over with old .env, uploads, user.css and that should be ok?

**Reply by ildyria** [reply]
are you using sqlite or mysql ?

You can always dump your db, load it locally on your machine, execute the migration, dump back and replace on your server.

**Reply by bussekas** [reply]
mysql is using right now. seems to be that some settings maybe main domain vs sub domain is not same because i just take main domain files and just put into sub domain and it works. at least i tested that there is buttons what is needed.

**Reply by ildyria** [reply]
Well that is good news. :)

**Reply by ildyria** [reply]
As you don't have ssh access, it makes it quite difficult to see what settings are wrong on your appache side. :(
